crates/common/src/paths.rs

```diff
@@ -36,6 +49,30 @@ pub fn resolve_manifest_file_path(provided_path: impl AsRef<Path>) -> Result<Pat
     }
 }
 
+/// Starting from the current directory, searches upward through
+/// the directory tree for a manifest (that is, a file with the default
+/// manifest name `spin.toml`). If found, the path to the manifest
+/// is returned, with a boolean flag indicating if the found path was
+/// the default (i.e. `spin.toml` in the current directory).
+/// If no matching file is found, the function returns None.
+pub fn search_upwards_for_manifest() -> Option<(PathBuf, bool)> {
+    let mut inferred_dir = std::env::current_dir().unwrap();
+    let mut is_default = true;
+
+    loop {
+        let candidate = inferred_dir.join(DEFAULT_MANIFEST_FILE);
+
+        if candidate.is_file() {
+            return Some((candidate, is_default));
+        }
+
+        is_default = false;
+        let parent = inferred_dir.parent()?;
+
+        inferred_dir = parent.to_owned();
+    }
```

Comment:
Do we want to restrict the level of parent directories that we look for? or would that be unexpected behavior.

Reply:
The one heuristic I would maybe consider would be to stop the search if you see a `.git` directory. I can imagine edge cases where traversing up out of a repository could cause some pain. :shrug:

Reply:
Okay, it now stops if it sees a `.git` directory, and I also changed the loop to top out at 20 iterations - this shouldn't stop normal use but it does provide belt and braces in case something weird is going on.  The 20 is arbitrary and can be increased or decreased if people have opinions.  "If" people have opinions, who am I kidding